File: android/app/src/main/kotlin/com/pixeldev/Docly/pdf/DrawingHandler.kt
package com.pixeldev.Docly.pdf

data class StrokePoint(
    val x: Float,
    val y: Float
)

data class StrokeOperation(
    val page: Int,
    val color: Int,
    val width: Float,
    val points: List<StrokePoint>
)

class DrawingHandler {
    private val operations = mutableListOf<StrokeOperation>()

    fun addStroke(page: Int, color: Int, width: Float, points: List<StrokePoint>) {
        if (points.size < 2) {
            return
        }
        operations += StrokeOperation(page = page, color = color, width = width, points = points)
    }

    fun getOperations(): List<StrokeOperation> = operations.toList()

    fun clear() {
        operations.clear()
    }
}

File: ios/Runner/pdf/DrawingHandler.swift
import Foundation

struct StrokePoint {
  let x: Double
  let y: Double
}

struct StrokeOperation {
  let page: Int
  let color: Int
  let width: Double
  let points: [StrokePoint]
}

final class DrawingHandler {
  private var operations: [StrokeOperation] = []

  func addStroke(page: Int, color: Int, width: Double, points: [StrokePoint]) {
    guard points.count > 1 else { return }
    operations.append(StrokeOperation(page: page, color: color, width: width, points: points))
  }

  func allOperations() -> [StrokeOperation] {
    operations
  }

  func clear() {
    operations.removeAll()
  }
}
